apis/python/src/tiledbsoma/_dask/load.py

```diff
@@ -0,0 +1,174 @@
+
+from tiledbsoma._dask.util import (
+    ChunkSize,
+    DaskConfig,
+    JoinIDs,
+    chunk_ids_sizes,
+    coords_to_joinids,
+    make_context,
+)
+from tiledbsoma._dense_nd_array import DenseNDArray
+from tiledbsoma._fastercsx import CompressedMatrix, Format
+from tiledbsoma._indexer import IntIndexer
+from tiledbsoma._sparse_nd_array import SparseNDArray
+from tiledbsoma.options._soma_tiledb_context import ConfigDict, ConfigVal
+
+if TYPE_CHECKING:
+    try:
+        import dask.array as da
+    except ImportError:
+        pass
+
+
+__all__ = [
+    # Re-export `DaskConfig` here, to save most users from needing separate `tiledbsoma._dask` import lines. It also
+    # appears in the public API of `SparseNDArrayRead.dask_array`, which imports it directly from
+    # `tiledbsoma._dask.util` (to avoid a circular dependency, since `SparseNDArray` is also referenced in this file).
+    "DaskConfig",
+    "load_daskarray",
+    "sparse_chunk",
+]
+
+
+def sparse_chunk(
+    block: npt.NDArray[tuple[JoinIDs, JoinIDs]],
+    block_info: dict[int | None, Any],
+    uri: str,
+    tiledb_config: ConfigDict,
+    tiledb_concurrency: int | None = None,
+    format: Format = "csr",
+    result_order: ResultOrderStr = ResultOrder.AUTO,
+    platform_config: PlatformConfig | None = None,
+) -> sp.csr_matrix | sp.csc_matrix:
+    """Load a slice of a TileDB-SOMA ``X`` matrix, as a block of a CSR-backed Dask Array."""
+    shape = block_info[None]["chunk-shape"]
+    assert block.shape == (1, 1)
+    joinids = block[0, 0]
+    obs_joinids, var_joinids = joinids
+    obs_joinids = obs_joinids.astype("int64")
+    var_joinids = var_joinids.astype("int64")
+    soma_ctx = make_context(
+        tiledb_config=tiledb_config,
+        tiledb_concurrency=tiledb_concurrency,
+    )
+    with SparseNDArray.open(uri, context=soma_ctx) as arr:
+        tbls = arr.read(
+            coords=(obs_joinids, var_joinids),
+            result_order=result_order,
+            platform_config=platform_config,
+        ).tables()
+        # TODO: i32-based reindexing?
+        obs_indexer = IntIndexer(obs_joinids, context=soma_ctx)
+        var_indexer = IntIndexer(var_joinids, context=soma_ctx)
+        new_tbls = []
+        for tbl in tbls:
+            new_dim0 = obs_indexer.get_indexer(tbl["soma_dim_0"]).astype("int32")
+            new_dim1 = var_indexer.get_indexer(tbl["soma_dim_1"]).astype("int32")
+            new_tbl = pa.Table.from_pydict(
+                {
+                    "soma_dim_0": new_dim0,
+                    "soma_dim_1": new_dim1,
+                    "soma_data": tbl["soma_data"],
+                }
+            )
+            new_tbls.append(new_tbl)
+        cm = CompressedMatrix.from_soma(
+            new_tbls,
+            shape=shape,
+            format=format,
+            make_sorted=True,
+            context=soma_ctx,
+        )
+        csx = cm.to_scipy()
+        return csx
+
+
+def load_daskarray(
```

Comment:
how do you pass timestamp info to the Dask loader? I.e., if the user is opening the Experiment in the past, does it get the same result as the non-Dask variant? (this would make a good unit test!)

Put another way, if `layer` was opened at a point in time, how is that information passed to the `sparse_chunk` worker?  It looks like it isn't passed, and `sparse_chunks` will open the "latest" variant, which I believe would be a bug.